src/bson/helpers/builder.cr

```diff
@@ -121,11 +121,41 @@ struct BSON
       @io.write_bytes 0x00_u8, IO::ByteFormat::LittleEndian
     end
 
+    def []=(key : String, value : UInt8)
+      field(:int32, key)
+      @io.write_bytes value.to_i32, IO::ByteFormat::LittleEndian
+    end
+
+    def []=(key : String, value : Int8)
+      field(:int32, key)
+      @io.write_bytes value.to_i32, IO::ByteFormat::LittleEndian
+    end
+
+    def []=(key : String, value : UInt16)
+      field(:int32, key)
+      @io.write_bytes value.to_i32, IO::ByteFormat::LittleEndian
+    end
+
+    def []=(key : String, value : Int16)
+      field(:int32, key)
+      @io.write_bytes value.to_i32, IO::ByteFormat::LittleEndian
+    end
+
+    def []=(key : String, value : UInt32)
+      field(:int32, key)
+      @io.write_bytes value.to_i32, IO::ByteFormat::LittleEndian
+    end
+
     def []=(key : String, value : Int32)
       field(:int32, key)
       @io.write_bytes value, IO::ByteFormat::LittleEndian
     end
 
+    def []=(key : String, value : UInt64)
+      field(:int64, key)
+      @io.write_bytes value.to_i64, IO::ByteFormat::LittleEndian
```

Comment:
I'm not sure this one makes sense, casting a `UInt64` to `Int64` definitely loses precision unless the user tracks down the value and casts it back when de-serializing data.

Reply:
Fixed in following commit : 26d497b85f33577faeea86a7c2519ae42afa8869